custom/conf/app.ini.sample

```diff
@@ -296,6 +296,9 @@ DEFAULT_KEEP_EMAIL_PRIVATE = false
 ; Default value for AllowCreateOrganization
 ; New user will have rights set to create organizations depending on this setting
 DEFAULT_ALLOW_CREATE_ORGANIZATION = true
+; Default value for EnableDependencies
+; Repositories will use depencies by default depending on this setting
+DEFAULT_ENABLE_DEPENDENCIES = true
```

Comment:
this should be off by default

Reply:
I would prefer this be on by default

Reply:
Mee too. I also think we'd need to be consistent, if we disable this by default we'd need to disable timetracking by default too.

Reply:
Too many toggles isn't good either. Issue tracking on/off per project I do understand. But if you wish to have issues, why would you want to turn of dependencies? Just don't use it then, like labels. No need to use them. Dependencies on and no ability to turn off. Unless it's not stable, but in that case it should be fixed instead of being toggleable

Reply:
Made a quick benchmark:
![bildschirmfoto vom 2018-01-02 13-41-09](https://user-images.githubusercontent.com/13721712/34484061-a07ec480-efc2-11e7-8fb4-dcf19308bab2.png)

Reply:
@ptman well you can still enable or disable them in repo settings. The only reason I could think of why you want to disable them is because of performance, but as the benchmark shows it doesn't have much of an impact if they are disabled or enabled...

Reply:
Every feature Gitea implement should not be turned off by default. That should be the highlight of the release. If some one don't want to use it, don't use it then. But to me every big/nice feature like this and time tracking and many more should be on by default. If benchmark is the scale, then i don't see any feature to be added to any application.

Reply:
@mmarif4u I totally agree with you. And its not very motivating if you create a feature and in the end when its very close to be merged and some people say "sorry but i think we should hide the feature by default so if anyone wants to use it he would need to first know it exists and then activly edit the configuration to enable and use it".
I think it's enough there is the possibility to disable it if one doesn't want to use it. And apparently (see the benchmark) it doesn't really have an impact on load times if it is enabled or disabled.

Reply:
I agree with @mmarif4u here, as long as there's no performance issue we shouldn't be disabling features. And to extend on it, I don't think that defaults should be in `app.ini` at all but rather stored in the database per User.